pkg/urbit/noun/imprison.c

```diff
@@ -313,12 +313,17 @@ u3i_word(c3_w dat_w)
 u3_atom
 u3i_chub(c3_d dat_d)
 {
-  c3_w dat_w[2] = {
-    dat_d & 0xffffffffULL,
-    dat_d >> 32
-  };
+  if ( c3y == u3a_is_cat(dat_d) ) {
```

Comment:
Nice. This works on doubles bc it's testing *any* set bit above 31 right?

Reply:
That's right:

```
#     define u3a_is_cat(som)    (((som) >> 31) ? c3n : c3y)
```